verisignmdns/src/main/java/denominator/verisignmdns/VerisignMDNSAllProfileResourceRecordSetApi.java

```diff
@@ -0,0 +1,127 @@
+        List<Record> recordList = api.getResourceRecordsList(domainName);
+        return VrsnContentConversionHelper.getSortedSetForDenominator(recordList).iterator();
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterateByName(String name) {
+        checkNotNull(name, "name was null");
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+    }
+
+    protected void put(Filter<ResourceRecordSet<?>> valid, ResourceRecordSet<?> rrset) {
+        checkNotNull(rrset, "rrset was null");
+        checkArgument(rrset.qualifier() != null, "no qualifier on: %s", rrset);
+        checkArgument(valid.apply(rrset), "%s failed on: %s", valid, rrset);
+
+        // @TODO IMPLEMENT -- in future development phase /////////
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+
+    }
+
+    @Override
+    public void put(ResourceRecordSet<?> rrset) {
+        // @ Keeping MOCK code for now...
+        // @TODO IMPLEMENT -- in future development phase /////////
+        put(notNull(), rrset);
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+    }
+
+    @Override
+    public Iterator<ResourceRecordSet<?>> iterateByNameAndType(String name, String type) {
+        checkNotNull(type, "type was null");
+        checkNotNull(name, "name was null");
+
+        List<Record> recordList = api.getResourceRecordsListForTypeAndName(domainName, type, name);
+        Iterator<ResourceRecordSet<?>> result = VrsnContentConversionHelper.getSortedSetForDenominator(recordList)
+                .iterator();
+        return result;
+    }
+
+    /**
+     * NOTE- for MDNS get only required ResourceRecordId ie. qualifier.
+     * Parameters name and type are ignored.
+     */
+    @Override
+    public ResourceRecordSet<?> getByNameTypeAndQualifier(String name, String type, String qualifier) {
+        ResourceRecordSet<?> result = null;
+        List<Record> recordList = api.getResourceRecordByQualifier(qualifier);
+        SortedSet<ResourceRecordSet<?>> rrSet = VrsnContentConversionHelper.getSortedSetForDenominator(recordList);
+        if (!rrSet.isEmpty()) {
+            result = rrSet.first();
+        }
+        return result;
+    }
+
+    @Override
+    public void deleteByNameTypeAndQualifier(String name, String type, String qualifier) {
+        // @ Keeping MOCK code for now...
+        // @TODO IMPLEMENT -- in future development phase /////////
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+
+    }
+
+    @Override
+    public void deleteByNameAndType(String name, String type) {
+        throw new VrsnMdnsException("Method Not Implemented", -1);
+    }
+
+    static class Factory implements denominator.AllProfileResourceRecordSetApi.Factory {
+        private Map<Zone, SortedSet<ResourceRecordSet<?>>> records;
+        private String domainName;
+        private VrsnMdns api;
+        private final Provider<Credentials> credentialsProvider;
```

Comment:
instead of passing credentials around everywhere, just apply it in the target as I mentioned in a prior review. 

ex. https://github.com/Netflix/denominator/blob/master/ultradns/src/main/java/denominator/ultradns/UltraDNSTarget.java